tests/src/bignum_codepath_check.c

```diff
@@ -0,0 +1,39 @@
+/** Support for path tracking in optionally safe bignum functions
+ */
+/*
+ *  Copyright The Mbed TLS Contributors
+ *  SPDX-License-Identifier: Apache-2.0 OR GPL-2.0-or-later
+ */
+
+#include "test/bignum_codepath_check.h"
+#include "bignum_core_invasive.h"
+
+#if defined(MBEDTLS_TEST_HOOKS) && !defined(MBEDTLS_THREADING_C)
+int mbedtls_codepath_check = MBEDTLS_MPI_IS_TEST;
```

Comment:
Now this is done with function pointers, the functions could take a pointer to an `int` to update, and that `int` could be on the stack in the test functions, meaning this could be thread-safe

Reply:
Hmm, no, maybe not, would still need to remember the pointer - ignore this